CRT.c

```diff
@@ -685,11 +689,94 @@ void CRT_restorePrivileges() {
 
 #endif /* HAVE_SETUID_ENABLED */
 
+#ifndef NDEBUG
+
+static int stderrRedirectNewFd = -1;
+static int stderrRedirectBackupFd = -1;
+
+static int createStderrCacheFile(void) {
+#ifdef HAVE_MEMFD_CREATE
+   return memfd_create("htop.stderr-redirect", 0);
+#else
+   char tmpName[] = "htop.stderr-redirectXXXXXX";
```

Comment:
In general its a good idea to set and restore the umask when using mkstemp, so here we'd insert code like:
```
mode_t cur_umask;
cur_umask = umask(S_IXUSR | S_IRWXG | S_IRWXO);
```

Reply:
[`mkstemp(3)`](https://manpages.debian.org/unstable/manpages-dev/mkstemp.3.en.html) creates the file with mode `0600`

Reply:
If you have an open FD with RW perms, wouldn't you want the actual file to actually be 0000, so noone else can access the file for neither reading nor writing?

Reply:
@cgzones see the man page section titled "NOTES" in mkstemp(3)...

```    
More generally, the POSIX specification of mkstemp() does not say  any‐
thing  about  file  modes, so the application should make sure its file
mode creation mask (see umask(2)) is set appropriately  before  calling
mkstemp() (and mkostemp()).
```

Reply:
added setting and re-setting umask around the mkstemp call